pkg/pot/ast.go

```diff
@@ -0,0 +1,143 @@
+package pot
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"golang.org/x/text/language"
+)
+
+type HeaderNode struct {
+	Language    language.Tag
+	Translator  string
+	PluralForms PluralForm
+}
+
+type PluralForm struct {
+	Plural   string
+	NPlurals int
+}
+
+type MessageNode struct {
+	MsgCtxt               string
+	MsgId                 string
+	MsgIdPlural           string
+	TranslatorComment     string
+	ExtractedComment      string
+	Reference             string
+	Flag                  string
+	MsgCtxtPrevCtxt       string
+	MsgIdPrevUntPluralStr string
+	MsgIdPrevUnt          string
+	MsgStr                []string
+}
+
+type Po struct {
+	Header   HeaderNode
+	Messages []MessageNode
+}
+
+func TokensToPo(tokens []Token) (Po, error) {
+	var messages []MessageNode
+
+	partsN := 2
+	currentMessage := MessageNode{}
+	header := HeaderNode{}
+
+	for _, token := range tokens {
+		switch token.Type {
+		case HeaderLanguage:
+			parts := strings.Split(token.Value, ": ")
```

Comment:
I have a feeling POT allows optional whitespace. Does it?

```suggestion
			parts := strings.Split(token.Value, ":")
			// both work
			parts := strings.Split(token.Value, ": ")
```

Reply:
Yes. you are right. PO files allow whitespaces in headers after the ":" character.